OpenRA.Mods.Common/Traits/Cargo.cs

```diff
@@ -144,7 +144,12 @@ void INotifyCreated.Created(Actor self)
 			conditionManager = self.Trait<ConditionManager>();
 		}
 
-		static int GetWeight(Actor a) { return a.Info.TraitInfo<PassengerInfo>().Weight; }
+		public int GetWeight(Actor a) { return GetPassengerTrait(a).Info.Weight; }
+
+		public Passenger GetPassengerTrait(Actor a)
```

Comment:
I see prolem with Cargo.GetWeight and Cargo.GetPipAt. How change Weight and PipType?
As a variant of first Passenger, but this seems a bad way.